assets/src/components/parts/janus-popup/Popup.tsx

```diff
@@ -209,6 +214,21 @@ const Popup: React.FC<PartComponentProps<PopupModel>> = (props) => {
     };
   }, [props.notify]);
 
+  useEffect(() => {
+    if (showPopup) {
```

Comment:
does this flag actually "prove" that the user did it? seems like we'd need a click event on a button or something?

Reply:
If this flag is true then it means the pop-up was opened and student had read it hence, I set the `userOpened `property to it.

If we  go with the button click then the problem with that it, it the pop-up is opened by default then students will only close the pop-up so the main pop-up button will never get clicked hence I didn't set it on button click.